agents/scripts/install-dependencies.sh

```diff
@@ -0,0 +1,146 @@
+#!/bin/bash
+set -e
+
+scripts_dir="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
+
+set -a
+source "$scripts_dir/config.env"
+source "$scripts_dir/${ENV_FILE}"
+set +a
+
+echo "[INFO] Updating and installing system packages..."
+apt-get clean && rm -rf /var/lib/apt/lists/* && apt-get update
+apt-get install -y build-essential zlib1g-dev wget git
+
+if [ "$INSTALL_FFI" = "true" ]; then
+    apt-get install -y libffi-dev
+fi
+
+if [ "$INSTALL_NCURSES" = "true" ]; then
+    apt-get install -y libncurses5-dev
+fi
+
+if [ "$INSTALL_GDBM" = "true" ]; then
+    apt-get install -y libgdbm-dev
+fi
+
+if [ "$INSTALL_OPEN_SSL" = "true" ]; then
+    apt-get install -y libssl-dev
+fi
+
+if [ "$INSTALL_READLINE" = "true" ]; then
+    apt-get install -y libreadline-dev
+fi
+
+if [ "$INSTALL_SQLITE" = "true" ]; then
+    apt-get install -y libsqlite3-dev
+fi
+
+mkdir -p /python-build/src
+cd /python-build/src
+
+if ! "${PYTHON_DIR}/bin/python" --version > /dev/null 2>&1; then
+    mkdir -p python
+
+    if [ ! -f python/python.tar.gz ]; then
+        echo "[INFO] Downloading Python 2.6.9..."
+        python_url="https://www.python.org/ftp/python/2.6.9/Python-2.6.9.tgz"
+        wget -O python/python.tar.gz ${python_url}
+        tar -C python --strip-components 1 -xzf python/python.tar.gz
+    fi
+
+    cd python
+
+    if [ "$BUILD_ZLIB_FROM_SRC" = "true" ]; then
+        echo "[INFO] Building zlib..."
+        cp Modules/Setup.dist Modules/Setup
+        sed -i '/zlibmodule\.c/ s/^# *//' Modules/Setup
+    fi
+
+    ncores="$(grep ^processor /proc/cpuinfo 2>/dev/null | wc -l | xargs)"
+    ncores="${numcores#0}"
```

Comment:
Correct typo in `ncores` assignment. You used `ncores` in the previous assignment but referenced `numcores` in the next line.

Reply:
Fixed.